tools/analyzer_plugin/test/integration/assists/toggle_statefulness_test.dart

```diff
@@ -0,0 +1,151 @@
+import 'dart:async';
+
+import 'package:analyzer_plugin/utilities/assist/assist.dart';
+import 'package:over_react_analyzer_plugin/src/assist/toggle_stateful.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/assist_test_base.dart';
+import 'boilerplate_assist_utils.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AddStatefulnessAssist);
+    defineReflectiveTests(RemoveStatefulnessAssist);
+  });
+}
+
+@reflectiveTest
+class AddStatefulnessAssist extends AssistTestBase with BoilerplateAssistTestStrings {
+  static int expectedPriority = ToggleComponentStatefulness.statefulAssistPriority;
+
+  @override
+  AssistKind get assistKindUnderTest => ToggleComponentStatefulness.makeStateful;
+
+  Future<void> test_noAssistOnFactory() async {
+    final generatedSource = simpleUiComponentSource();
+
+    var source = newSource(BoilerplateAssistTestStrings.fileName, generatedSource);
```

Comment:
#nit is having this long constant necessary? I would say just put in a dummy name, or make `fileName` an instance member so you don't have to prefix it
```suggestion
    var source = newSource(fileName, generatedSource);
```

Reply:
We could also just make newSource's fileName arg be an optional/named arg, since a lot of the time you don't beed it
```suggestion
    var source = newSource(generatedSource);
```

Reply:
Good call, I can shorten that! We could make it optional, but I think that's a larger question than this PR since the method is in use elsewhere (and those spots would have to be updated too)

Reply:
Oh yeah for sure, we could definitely address that separately; just thought it'd be worth mentioning that these helper methods are still fairly new and we should feel free to change em to better suit our use cases!

Reply:
True true, that's a good point!